frontend/src/Initialize.tsx

```diff
@@ -216,6 +217,188 @@ const Initialize = (props: Props) => {
+    evt.preventDefault();
+    localStorage.setItem('gameSettingsInLocalStorage', JSON.stringify(props.state.propagated));
+  };
+
+  const loadGameSettings = (evt: any) => {
+    evt.preventDefault();
+    const settings = localStorage.getItem('gameSettingsInLocalStorage');
+    if (settings !== null) {
+      const gameSettings: IPropagatedState = JSON.parse(settings);
+      for (const [key, value] of Object.entries(gameSettings)) {
+        switch (key) {
+          case 'game_mode':
+            if (value === 'Tractor') {
+              send({
+                Action: {
+                  SetGameMode: 'Tractor'
+                }
+              });
+            } else {
+              send({
+                Action: {
+                  SetGameMode: {
+                    FindingFriends: {
+                      num_friends: value.num_friends,
+                    },
+                  },
+                },
+              });
+            }
+            break;
+          case 'num_decks':
+            send({
+              Action: {
+                SetNumDecks: value
+              },
+            })
+            break;
+          case 'kitty_size':
+            send({
+              Action: {
+                SetKittySize: value,
+              },
+            });
+            break;
+          case 'friend_selection_policy':
+            send({
+              Action: {
+                SetFriendSelectionPolicy: value,
+              },
+            });
+            break;
+          case 'hide_landlord_points':
+            send({
+              Action: {
+                SetHideLandlordsPoints: value
+              },
+            });
+            break;
+          case 'hide_played_cards':
+            send({ Action: { SetHidePlayedCards: value } });
+            break;
+          case 'advancement_policy':
+            send({
+              Action: {
+                SetAdvancementPolicy: value,
+              },
+            });
+            break;
+          case 'kitty_bid_policy':
+            send({
+              Action: {
+                SetKittyBidPolicy: value,
+              },
+            });
+            break;
+          case 'kitty_penalty':
+            send({
+              Action: {
+                SetKittyPenalty: value,
+              },
+            });
+            break;
+          case 'throw_penalty':
+            send({
+              Action: {
+                SetThrowPenalty: value,
+              },
+            });
+            break;
+          case 'trick_draw_policy':
+            send({
+              Action: {
+                SetTrickDrawPolicy: value,
+              },
+            });
+            break;
+          case 'throw_evaluation_policy':
+            send({
+              Action: {
+                SetThrowEvaluationPolicy: value,
+              },
+            });
+            break;
+        }
+      }
+    }
+  }
+
+  const resetGameSettings = (evt: any) => {
```

Comment:
Hm. Rather than having a `resetGameSettings` which requires us to hardcode all of the default values, what do you think adding an endpoint that returns a default `PropagatedState`? e.g. at /default_settings.

We could then XHR request it, and the code you've already written for loading game settings from a serialized propagated state.

Reply:
sounds good since you already have the default settings in the backend